### PR TITLE
Update the doc for kyma.2.0

### LINKED ISSUE #11746: [Kyma 2.0] Rework API Gateway documentation
**Description**

Review API Gateway documentation located on the `kyma-2.0-docu branch` - Goat SME's help required. Check if the documents are valid and accurate. 

- [ ] Update the documents to reflect changes in the API Gateway components made for Kyma 2.0 release.
- [ ] Remove outdated information.
- [ ] Update diagrams.
- [ ] Make sure the documentation is compliant with the [Content Strategy K2.0](https://deploy-preview-565--kyma-project-community-preview.netlify.app/guidelines/content?utm_source=github&utm_campaign=bot_dp#content-strategy-k-2-0-content-strategy-k-2-0).
- [ ] Add missing documents (overviews, troubleshooting).

**Area**

- API Gateway

**Reasons**

We need accurate and up-to-date documentation for Kyma 2.0 release.

**Assignees**

@mmitoraj 
@strekm

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Update links
- Update diagrams
- Update content
- Update structure

**Related issue(s)**
Resolves #11746

### CODE REVIEW

**Comment by mmitoraj** [comment]
API EXPOSURE
1. Overview -> Main areas -> API Exposure
2. Overview -> Main areas -> API Exposure -> API Gateway
3. Overview -> Main areas -> API Exposure -> Useful links
4. Tutorials -> API Exposure -> Expose a service
5. Tutorials -> API Exposure -> Expose and secure a service
6. Tutorials -> API Exposure -> Use a custom domain to expose service
7. Operation guides -> Troubleshooting -> Cannot connect to a service exposed by an APIRule
8. Operation guides -> Troubleshooting -> External DNS management - connection refused or timeout
9. Operation guides -> Troubleshooting -> External DNS management - could not resolve host
10. Operation guides -> Troubleshooting -> External DNS management - resource ignored by the controller
11. Operation guides -> Troubleshooting -> Certificate management - Issuer not created
12. Technical reference -> Allowed domains in the API Gateway Controller
13. Technical reference -> Blocked services in the API Gateway Controller
14. Technical reference -> Configuration parameters -> ORY chart
15. Technical reference -> Custom resources -> APIRule
16. Technical reference -> Configuration parameters -> Istio chart

**Review by strekm — DISMISSED** [review]
/meow

File: docs/05-technical-reference/apix-02-whitelisted-domains.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ The API Gateway Controller uses an allowlist of domains for which it allows to e
 
 If the domain does not match the allowlist, the API Gateway Controller sets an appropriate validation status on the APIRule CR created for that service.
 
->**TIP:** For more information, read about the [Api CR statuses](#custom-resource-api-rule-status-codes).
+>**TIP:** For more information, read about the [Api CR statuses](./00-custom-resources/apix-01-apirule.md#status-codes).
```
Comment (NHingerl) [inline docs/05-technical-reference/apix-02-whitelisted-domains.md:9]:
```suggestion
>**TIP:** For more information, read about the [API CR statuses](./00-custom-resources/apix-01-apirule.md#status-codes).
```

File: docs/05-technical-reference/apix-03-blacklisted-services.md [hunk 2]
```diff
@@ -4,6 +4,6 @@ title: Blocked services in the API Gateway Controller
 
 The API Gateway Controller uses a blocklist of services for which it does not create either a Virtual Service or Oathkeeper Access Rules. As a result, these services cannot be exposed. Every time a user creates a new APIRule custom resource (CR) for a service, the API Gateway Controller checks the name of the service specified in the CR against the blocklist. If the name of the service matches a blocklisted entry, the API Gateway Controller sets an appropriate validation status on the APIRule CR created for that service.
 
->**TIP:** For more information, read about the [Api CR statuses](#custom-resource-api-rule-status-codes).
+>**TIP:** For more information, read about the [Api CR statuses](./00-custom-resources/apix-01-apirule.md#status-codes).
```
Comment (NHingerl) [inline docs/05-technical-reference/apix-03-blacklisted-services.md:7]:
```suggestion
>**TIP:** For more information, read about the [API CR statuses](./00-custom-resources/apix-01-apirule.md#status-codes).
```

File: docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- Learn something more about:
+
+  - [Authorization configuration](../../../05-technical-reference/apix-01-config-authorizations-apigateway.md)
+  - [Allowed domains in the API Gateway Controller](../../../05-technical-reference/apix-02-whitelisted-domains.md)
+  - [Blocked services in the API Gateway Controller](../../../05-technical-reference/apix-03-blacklisted-services.md)
+
+- Analyze configuration details for:
+
+  - The [ORY chart](../../../05-technical-reference/05-configuration-parameters/apix-02-ory-chart.md)
```
Comment (NHingerl) [inline docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md:28]:
```suggestion
  - The [ORY chart](../../../05-technical-reference/00-configuration-parameters/apix-02-ory-chart.md)
```
link was broken

**Review by NHingerl — CHANGES_REQUESTED** [review]

**Comment by mmitoraj** [comment]
Invalid commits after solving a merge conflict.

**Pull request closed without merging**